json.c

```diff
@@ -228,7 +228,7 @@ static const uint8_t state_transition_table[NR_STATES][NR_CLASSES] = {
 /*N2*/ PT_(__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,N3,__,__,__,__,__,__,__,__,__,__),
 /*N3*/ PT_(__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,NU,__,__,__,__,__,__,__,__,__,__),
 /****************************************************************************************************************/
-/*C1*/ PT_(__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,C2,__),
+/*C1*/ PT_(__,__,__,__,__,__,__,__,__,__,__,YB,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,__,C2,__),
```

Comment:
Although moving from C1 to YB produces a correct parsing, there are a couple of operations performed on this transition that introduce some very minor inconsistencies:
- allow_yaml_comments needs to be true to allow such comments
- buffer is needlessly flushed 

you might obtain the correct behavior by simply transition from C1 to Y1 given the fact that all operations needed at the beginning of a comment were already performed when entering the CB state.

We might also rename Y1 state to reflect the fact that not only YAML comments are entering it but all one line comments.